common/src/main/java/org/wso2/testgrid/common/util/S3StorageUtil.java

```diff
@@ -50,13 +65,54 @@ public static String getS3BucketURL() {
         return String.join("/", TestGridConstants.AMAZON_S3_URL, s3BucketName);
     }
 
+    public static void deleteTestPlan(TestPlan testPlan) {
+
+        try {
+            ArtifactReadable artifactDownloadable = new AWSArtifactReader(ConfigurationContext.
+                    getProperty(ConfigurationContext.ConfigurationProperties.AWS_REGION_NAME),
+                    ConfigurationContext.getProperty(ConfigurationContext.ConfigurationProperties.AWS_S3_BUCKET_NAME));
+
+            String s3KeyName = deriveS3TestPlanDirPath(testPlan, artifactDownloadable);
+            logger.info("Files cleaned in :" + s3KeyName);
+            String clientRegion = ConfigurationContext.getProperty
+                    (ConfigurationContext.ConfigurationProperties.AWS_REGION_NAME);
+            String bucketName = ConfigurationContext.getProperty
+                    (ConfigurationContext.ConfigurationProperties.AWS_S3_BUCKET_NAME);
+
+            AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
+                    .withCredentials(new ProfileCredentialsProvider())
+                    .withRegion(clientRegion)
+                    .build();
+
+            ObjectListing objectListing = s3Client.listObjects(bucketName, s3KeyName);
+            List<S3ObjectSummary> objectSummaries = objectListing.getObjectSummaries();
+
+            for (S3ObjectSummary os : objectSummaries) {
+                logger.info(StringUtil.concatStrings("deleting file", os.getKey(), "of test plan ",
+                        testPlan.getId()));
+                s3Client.deleteObject(new DeleteObjectRequest(bucketName, os.getKey()));
+            }
+
+            s3Client.deleteObject(new DeleteObjectRequest(bucketName, s3KeyName));
+
+        } catch (AmazonServiceException e) {
+            logger.error("Amazon S3 couldn't process request ", e);
+        } catch (SdkClientException e) {
+            logger.error("Amazon S3 couldn't be contacted for a response" , e);
+        } catch (ArtifactReaderException e) {
+            logger.error("Error occurred when reading the artifact.", e);
```

Comment:
Missing context. You need to log for which test-plan this issue occurred, and any other useful information.

Suppose, you have a 1000MB log file where this issue was there in the middle. Think how much context information you need to tackle down and fix the issue or identify the affects of it.